### PR TITLE
chore(region-info): register eu-central-2, fix firehose cidr blocks for ap-south-2, eu-south-2, me-central-1

### LINKED ISSUE #24477: Register new region: eu-central-2
🚀 Checklist:

- [x] Add region in [AWS_CDK_METADATA](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L1) (commented out).
- [x] Add region and zone id in [ROUTE_53_BUCKET_WEBSITE_ZONE_IDS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L40).
- [ ] Add region and zone id in [EBS_ENV_ENDPOINT_HOSTED_ZONE_IDS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L77).
- [ ] Add region and account in [ELBV2_ACCOUNTS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L116).
- [ ] Add region and account in [DLC_REPOSITORY_ACCOUNTS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L149).
- [ ] Add region and account in [APPMESH_ECR_ACCOUNTS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L174).
- [x] Add region and ARN in [CLOUDWATCH_LAMBDA_INSIGHTS_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L202) for 1.0.178.0.
- [x] Add region and CIDR block address in [FIREHOSE_CIDR_BLOCKS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L498).
- [x] Add region and ARN in [ADOT_LAMBDA_LAYER_JAVA_SDK_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L529) for most recent version in x86_64 and arm64.
- [x] Add region and ARN in [ADOT_LAMBDA_LAYER_JAVA_AUTO_INSTRUMENTATION_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L608) for most recent version in x86_64 and arm64.
- [x] Add region and ARN in [ADOT_LAMBDA_LAYER_JAVASCRIPT_SDK_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L689) for most recent version in x86_64 and arm64.
- [x] Add region and ARN in [ADOT_LAMBDA_LAYER_PYTHON_SDK_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L768) for most recent version in x86_64 and arm64.
- [x] Add region and ARN in [ADOT_LAMBDA_LAYER_GENERIC_ARNS](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/build-tools/fact-tables.ts#L847) for most recent version in x86_64 and arm64.
- [x] Add region in [AWS_REGIONS_AND_RULES](https://github.com/aws/aws-cdk/blob/v2.65.0/packages/@aws-cdk/region-info/lib/aws-entities.ts).
- [ ] Run integ tests and update snapshots

### PR DESCRIPTION
**Pull request by sumupitchayan**
Checks off some items from #24477

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by otaviomacedo — APPROVED** [review]

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-iot-actions-alpha/test/kinesis-firehose/integ.firehose-put-record-action.js.snapshot/test-stack.template.json [hunk 1]
```diff
@@ -273,7 +273,7 @@
     "FirehoseCidrBlock": "35.158.127.160/27"
    },
    "eu-central-2": {
-    "FirehoseCidrBlock": "16.62.183.32/27"
+    "FirehoseCidrBlock": "16.62.183.64/26/27"
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-iot-actions-alpha/test/kinesis-firehose/integ.firehose-put-record-action.js.snapshot/test-stack.template.json:276]: This doesn't look right

Comment (mrgrain) [reply packages/@aws-cdk/aws-iot-actions-alpha/test/kinesis-firehose/integ.firehose-put-record-action.js.snapshot/test-stack.template.json:276]: Same for `eu-south-2` by the way, although that's not your PR

File: packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts [hunk 2]
```diff
@@ -29,8 +29,8 @@ export const AWS_CDK_METADATA = new Set([
   'me-south-1',
   // 'me-central-1',
   'sa-east-1',
-  // 'ap-south-2',
   // 'eu-central-2',
+  // 'ap-south-2',
```
Comment (mrgrain) [inline packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts:33]: (Non-blocking) Side note: This change is a noop. We probably should order that list alphabetically.... 🙈

File: packages/@aws-cdk/aws-iot-actions-alpha/test/kinesis-firehose/integ.firehose-put-record-action.js.snapshot/test-stack.template.json (hunk 1 above)
Comment (mrgrain) [reply packages/@aws-cdk/aws-iot-actions-alpha/test/kinesis-firehose/integ.firehose-put-record-action.js.snapshot/test-stack.template.json:276]: And `me-central-1` and a bunch of others. I think we need to refactor how the firehose cidr block fact works.

File: packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md
Comment (mrgrain) [inline packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md:1]: Revert this change, it's unrelated.

File: packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts [hunk 3]
```diff
@@ -533,12 +533,12 @@ export const FIREHOSE_CIDR_BLOCKS: { [region: string]: string } = {
...
   'eu-north-1': '13.53.63.224',
   'eu-south-1': '15.161.135.128',
-  'eu-south-2': '18.100.194.0/26',
+  'eu-south-2': '18.100.71.96',
   'eu-west-1': '52.19.239.192',
   'eu-west-2': '18.130.1.96',
   'eu-west-3': '35.180.1.96',
   'me-south-1': '15.185.91.0',
-  'me-central-1': '3.28.159.64/26',
+  'me-central-1': '15.185.91.0',
```
Comment (mrgrain) [inline packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts:541]: According to the link this should be `3.28.159.32`

**Review by mrgrain — APPROVED** [review]

**Pull request merged**